lib/mongo/crypt/status.rb

```diff
@@ -0,0 +1,125 @@
+## Copyright (C) 2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'ffi'
+
+module Mongo
+  module Crypt
+
+    # A wrapper around mongocrypt_status_t, representing the status of
+    # a mongocrypt_t handle.
+    #
+    # @since 2.12.0
+    class Status
+      # Create a new Status object
+      #
+      # @since 2.12.0
+      def initialize
+        @status = Binding.mongocrypt_status_new
+      end
+
+      # Set a label, code, and message on the Status
+      #
+      # @param [ Symbol ] label One of :ok, :error_client, or :error_kms
+      # @param [ Integer ] code
+      # @param [ String ] message
+      #
+      # @return [ Status ] returns self
+      #
+      # @since 2.12.0
+      def update(label, code, message)
```

Comment:
I might include these params in the initializer instead, but I'll wait to see how this is used.